modules/tinymce/src/plugins/link/test/ts/browser/ContextToolbarTest.ts

```diff
@@ -43,6 +47,22 @@ describe('browser.tinymce.plugins.link.ContextToolbarTest', () => {
     await UiFinder.pWaitForState<HTMLInputElement>('check link content', SugarBody.body(), '.tox-toolbar input', (ele) => ele.dom.value === 'http://www.google.com');
   });
 
+  it('TINY-8940: toolbar doesn\'t show on non link elements', () => {
+    const editor = hook.editor();
+    editor.options.set('link_context_toolbar', true);
+    editor.setContent('<p>google</p>');
+    Mouse.trueClickOn(TinyDom.body(editor), 'p');
+    // You might be wondering why I did not use UiFinder.notExists.
+    // And the reason being simply it doesn't work in this case.
+    // I am not sure why, but you can put a  link in the editor content
+    // and click on it. The tollbar will show as it should, while the call
+    // UiFinder.notExists(SugarBody.body(), '.tox-toolbar') will still pass
+    // when it should fail since the toolbar does exist
```

Comment:
I'm not sure what you did here, but I tested this locally and `notExists` does work if I revert your changes to `Controls.ts`. Additionally this is the exact same implementation of `UiFinder.notExists`, so it won't be helping anyways/will have the same issues. Here's what I used if it helps:
```ts
    await Waiter.pWait(50); // Need to wait a little for the toolbar to trigger
    UiFinder.notExists(SugarBody.body(), '.tox-pop__dialog .tox-toolbar');
```

Now the couple of things you'll note are different are that I added the `.tox-pop__dialog` part to the selector because we want to be sure it's the context toolbar that doesn't show and the toolbar class alone can be used elsewhere. The second thing I added was the wait, as the context toolbar does **not** trigger synchronously... it has a `17ms` timeout/throttle as can be seen here, which is potentially why you thought `notExists` wasn't working: https://github.com/tinymce/tinymce/blob/236891ffcdf2fd1b698d020e67c31838fb110019/modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbar.ts#L244

Reply:
That seemed to do the trick 👍